server/server.js

```diff
@@ -0,0 +1,115 @@
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts";
+mongoose
+  .connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+  .catch((error) => {
+    console.log(`Cannot connect to database: ${error}`);
+  });
+mongoose.Promise = Promise;
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(bodyParser.json());
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res
+      .status(500)
+      .json({ error: "Service unavailable" });
+  }
+});
```

Comment:
Wow! Learned something new here. Nice way of handling an error with the service! 🤩